src/Raven.Server/ServerWide/Commands/Indexes/PutAutoIndexCommand.cs

```diff
@@ -15,23 +15,26 @@ namespace Raven.Server.ServerWide.Commands.Indexes
 {
     public class PutAutoIndexCommand : UpdateDatabaseCommand
     {
+        public DateTime CreatedAt { get; set; }
         public AutoIndexDefinition Definition;
 
         public PutAutoIndexCommand()
         {
         }
 
-        public PutAutoIndexCommand(AutoIndexDefinition definition, string databaseName, string uniqueRequestId)
+        public PutAutoIndexCommand(AutoIndexDefinition definition, string databaseName, string uniqueRequestId, DateTime? createdAt = null)
             : base(databaseName, uniqueRequestId)
         {
             Definition = definition;
+            CreatedAt = createdAt ?? DateTime.MinValue;
         }
 
         public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
         {
             try
             {
-                record.AddIndex(Definition);
+                var setting = IndexStore.GetGlobalRollingSetting(record);
+                record.AddIndex(Definition, CreatedAt, etag, setting);
```

Comment:
Old commands will have DateTime.MinValue, it looks like it is fine, but double-checking.